### PR TITLE
change loggingconfig layout to include relevant trace/span info

### PR DESCRIPTION
**Pull request by AndrewWang996**

### CODE REVIEW

File: soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java [hunk 1]
```diff
@@ -20,7 +20,10 @@
    * <pattern>%-5p [%X{channel}%X{userId}%X{dsId}%X{txnId}%date{"yyyy-MM-dd HH:mm:ss,SSS Z"}] - %c{10}: %m%n%rEx</pattern>
    * documented here: http://logback.qos.ch/manual/layouts.html
    */
-  private String pattern = "%date{\"yyyy-MM-dd HH:mm:ss,SSS Z\"} %-5level %40.40(%logger{36}:%line) -%X{userId}%X{dsId}%X{client}%X{txnId}%X{resourceHash}%X{experimentsApplied} %message%n%rootException";
+//  private String pattern = "%date{\"yyyy-MM-dd HH:mm:ss,SSS Z\"} %-5level %40.40(%logger{36}:%line) -%X{userId}%X{dsId}%X{client}%X{txnId}%X{resourceHash}%X{experimentsApplied} %message%n%rootException";
```
Comment (relateiq-ec2) [inline soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java [hunk 2]
```diff
@@ -20,7 +20,10 @@
...
+  private String pattern = "%date{\"yyyy-MM-dd HH:mm:ss,SSS Z\"} %-5level %40.40(%logger{36}:%line) -%X{userId} %X{dsId} %X{client} t.%X{txnId} s.%X{spanId} p.%X{parentId} %X{annotation} %X{ipv4} %X{serviceName} %X{resourceHash}%X{experimentsApplied} %message%n%rootException";
```
Comment (mjburghoffer) [inline soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java]: if txnId already has a prefix of `txn=`, then no need for a `t.` as prefix

Comment (jontg) [inline soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java]: Looks like `txn=` is already prefixed [in the logging requestlogfilter](https://github.com/relateiq/soa/blob/299c9bb6b9496f0235a2a0a990299506b4a20231/soa-commons/src/main/java/com/relateiq/soa/commons/MDCTxnFilter.java#L21), as Burg suggests.

**Comment by AndrewWang996** [comment]
@mjburghoffer does this look good?

**Comment by relateiq-ec2** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by pinnamur** [comment]
LGTM

File: soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java [hunk 3]
```diff
@@ -20,7 +20,9 @@
    * <pattern>%-5p [%X{channel}%X{userId}%X{dsId}%X{txnId}%date{"yyyy-MM-dd HH:mm:ss,SSS Z"}] - %c{10}: %m%n%rEx</pattern>
    * documented here: http://logback.qos.ch/manual/layouts.html
    */
-  private String pattern = "%date{\"yyyy-MM-dd HH:mm:ss,SSS Z\"} %-5level %40.40(%logger{36}:%line) -%X{userId}%X{dsId}%X{client}%X{txnId}%X{resourceHash}%X{experimentsApplied} %message%n%rootException";
+  private String pattern = "%date{\"yyyy-MM-dd HH:mm:ss,SSS Z\"} %-5level %40.40(%logger{36}:%line) -%X{userId}%X{dsId}%X{client}%X{txnId}%X{spanId}%X{parentSpanId}%X{annotation}%X{ipv4}%X{serviceName}%X{resourceHash}%X{experimentsApplied} %message%n%rootException";
```
Comment (ankushgoyal) [inline soa-logging/src/main/java/com/relateiq/soa/logging/LoggingConfig.java]: 1. lets use resourceHash and remove serviceName
2. lets remove ipv4 & annotation

**Pull request merged**